File: packages/shared-types/src/assistant.ts
// Types for the AI assistant request/response.
export type IntentType =
  | "ask_next_step"
  | "ask_route"
  | "ask_current_status"
  | "out_of_scope_medical"
  | "unknown";

export type AssistantRequest = {
  message: string;
  session_id: string | null; // optional: some questions need no session
};

export type ChatbotMetadata = {
  current_step?: string | null;
  target_room?: string | null;
  target_floor?: string | null;
  next_action?: string | null;
  accessibility_mode?: string | null;
};

export type ChatbotMessageRequest = {
  session_id: string;
  sender_id: string;
  text: string;
  metadata?: ChatbotMetadata;
};

export type SmartBotCard = {
  type?: string;
  text?: string;
  title?: string;
  description?: string;
  content?: string;
  buttons?: Array<Record<string, unknown>>;
  audio_url?: string;
  play_type?: string;
  [key: string]: unknown;
};

export type ChatbotMessageResponse = {
  reply_text: string;
  cards: SmartBotCard[];
  intent_name: string | null;
  handoff_required: boolean;
};

export type AssistantResponse = {
  intent: IntentType;
  response_text: string;
  confidence: number;
  is_fallback: boolean;
  // When the answer points at a room, these help the UI offer a "chỉ đường" button.
  target_location_id: string | null;
  target_room: string | null;
};

File: services/navigation-engine/app/models/assistant.py
"""Assistant request/response models (mirror shared-types/assistant.ts)."""

from __future__ import annotations

from typing import Literal, Optional

from pydantic import BaseModel

IntentType = Literal[
    "ask_next_step",
    "ask_route",
    "ask_current_status",
    "out_of_scope_medical",
    "unknown",
]


class AssistantRequest(BaseModel):
    message: str
    session_id: Optional[str] = None  # optional: some questions need no session


class AssistantResponse(BaseModel):
    intent: str
    response_text: str
    confidence: float
    is_fallback: bool = False
    # When the answer points at a room, these help the UI offer a "chỉ đường" button.
    target_location_id: Optional[str] = None
    target_room: Optional[str] = None
